api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -146,19 +205,21 @@ private Order saveOrder(Order order, OrderContext orderContext, boolean isRetros
 		if (isDrugOrder) {
 			((DrugOrder) order).setAutoExpireDateBasedOnDuration();
 		}
-		
 		if (concept == null) {
 			throw new MissingRequiredPropertyException("Order.concept.required");
 		}
-		
+
+	}
+
+	private void ensureOrderTypeIsSet(Order order, OrderContext orderContext) {
 		Order previousOrder = order.getPreviousOrder();
```

Comment:
move `previousOrder` declaration to right before its first use. see that we declare it here but there are a few cases where we might return before the variable is actually used.